extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php

```diff
@@ -119,17 +125,20 @@ public function content( $params ){
 		}
 		$trendingArticles = [];
 		if ( !empty( $trendingArticlesData ) ) {
-			$items = array_slice( $trendingArticlesData, 0, self::MAX_TRENDING_ARTICLES );
+			$items = array_slice( $trendingArticlesData, 0, static::MAX_TRENDING_ARTICLES );
 			//load data from response to template
 			foreach( $items as $item ) {
 				$img = $this->app->sendRequest( 'ImageServing', 'getImages', [
```

Comment:
you can use `ImageServing` class directly. It allows you to fetch data for many articles at once.